src/query/functions/linear/round.go

```diff
@@ -24,67 +24,58 @@ import (
 	"fmt"
 	"math"
 
-	"github.com/m3db/m3/src/query/executor/transform"
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/functions/lazy"
+	"github.com/m3db/m3/src/query/parser"
 )
 
-// RoundType rounds each value in the timeseries to the nearest integer.
+// RoundType rounds each datapoint in the series.
 // Ties are resolved by rounding up. The optional to_nearest argument allows
-// specifying the nearest multiple to which the timeseries values should be rounded (default=1).
-// This multiple may also be a fraction.
+// specifying the nearest multiple to which the timeseries values should be
+// rounded (default=1). This variable may be a fraction.
+// Special cases are: round(_, 0) = NaN
 const RoundType = "round"
 
 type roundOp struct {
 	toNearest float64
 }
 
-// NewRoundOp creates a new round op based on the type and arguments
-func NewRoundOp(args []interface{}) (BaseOp, error) {
+func parseArgs(args []interface{}) (float64, error) {
 	if len(args) > 1 {
-		return emptyOp, fmt.Errorf("invalid number of args for round: %d", len(args))
+		return 0, fmt.Errorf("invalid number of args for round: %d",
+			len(args))
 	}
 
-	var (
-		toNearest = 1.0
-		ok        bool
-	)
-	if len(args) > 0 {
-		toNearest, ok = args[0].(float64)
-		if !ok {
-			return emptyOp, fmt.Errorf("unable to cast to to_nearest argument: %v", args[0])
+	if len(args) == 1 {
+		if nearest, ok := args[0].(float64); ok {
+			return nearest, nil
 		}
-	}
 
-	spec := roundOp{
-		toNearest: toNearest,
+		return 0, fmt.Errorf("unable to cast to to_nearest argument: %v",
+			args[0])
 	}
 
-	return BaseOp{
-		operatorType: RoundType,
-		processorFn:  makeRoundProcessor(spec),
-	}, nil
-
+	return 1, nil
```

Comment:
Interesting, why does args > 1 always return a 1 value?  Should this be error instead or no?

Reply:
So if there's more than 1 arg it will error out, 1 arg will attempt to parse the float, and this path is for 0 args which just returns the default value; will add a comment explaining this 👍